client/ayon_core/pipeline/context_tools.py

```diff
@@ -579,3 +579,14 @@ def get_process_id():
     if _process_id is None:
         _process_id = str(uuid.uuid4())
     return _process_id
+
+
+def version_up_workfile():
+    """Function to increment and save workfile
+    """
+    host = registered_host()
+    current_file = host.get_current_workfile()
+    if not current_file:
+        return None
+    filepath = version_up(current_file)
+    host.save_workfile(filepath)
```

Comment:
The question now becomes - can we also make it save the workfile with a default name if it wasn't saved? Because it still seems this may not be using the exact same logic the workfiles tool would?

Reply:
I would like to know how to do it. @iLLiCiTiT do you have any idea?

Reply:
I guess to code for it is here https://github.com/ynput/ayon-core/blob/750ad82744145da0397100fef98c04d4e150bcee/client/ayon_core/tools/workfiles/models/workfiles.py#L381-L439